pkg/cli/admin/compare/compare_test.go

```diff
@@ -0,0 +1,311 @@
+			mode: []Mode{Live, Local},
+		},
+		{
+			name: "Required Resources Of Optional Component Are Reported Missing If At Least One Of Resources In Group Is Included",
+			mode: []Mode{Live, Local},
+		},
+	}
+	tf := cmdtesting.NewTestFactory()
+	testFlags := flag.NewFlagSet("test", flag.ContinueOnError)
+	klog.InitFlags(testFlags)
+	klog.LogToStderr(false)
+	_ = testFlags.Parse([]string{"--skip_headers"})
+	for _, test := range tests {
+		for _, mode := range test.mode {
+			t.Run(test.name+string(mode), func(t *testing.T) {
+				IOStream, _, out, _ := genericiooptions.NewTestIOStreams()
+				klog.SetOutputBySeverity("INFO", out)
+				cmd := getCommand(t, &test, mode, tf, &IOStream)
+				cmdutil.BehaviorOnFatal(func(str string, code int) {
+					getGoldenValue(t, path.Join(test.getTestDir(), fmt.Sprintf("%serr.golden", mode)), []byte(str+string(rune(code))))
+					panic("Expected Error Test Case")
+				})
+				defer func() {
+					_ = recover()
+					getGoldenValue(t, path.Join(test.getTestDir(), fmt.Sprintf("%sout.golden", mode)), removeInconsistentInfo(out.String()))
+				}()
+				cmd.Run(cmd, []string{})
+			})
+		}
+	}
+}
+
+func removeInconsistentInfo(text string) []byte {
+	//remove diff tool generated temp directory path
+	re := regexp.MustCompile("\\/tmp\\/(?:LIVE|MERGED)-[0-9]*")
+	text = re.ReplaceAllString(text, "TEMP")
+	//remove diff datetime
+	re = regexp.MustCompile("(\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}\\.\\d{9} [+-]\\d{4})")
+	return []byte(re.ReplaceAllString(text, "DATE"))
+}
+
+func getGoldenValue(t *testing.T, fileName string, value []byte) {
+	if *update {
+		t.Log("update golden file")
+		if err := os.WriteFile(fileName, value, 0644); err != nil {
+			t.Fatalf("test %s failed to update golden file: %s", fileName, err)
+		}
+	}
+	expected, err := os.ReadFile(fileName)
+	if err != nil {
+		t.Fatalf("test %s failed reading .golden file: %s", fileName, err)
+	}
+	require.Equal(t, string(expected), string(value))
+	return
+}
+
+func getCommand(t *testing.T, test *Test, mode Mode, tf *cmdtesting.TestFactory, streams *genericiooptions.IOStreams) *cobra.Command {
+	cmd := NewCmd(tf, *streams)
+	require.NoError(t, cmd.Flags().Set("concurrency", defaultConcurrency))
+	if !test.leaveTemplateDirEmpty {
+		require.NoError(t, cmd.Flags().Set("reference", path.Join(test.getTestDir(), TestRefDirName)))
+	}
+	if test.shouldDiffAll {
+		require.NoError(t, cmd.Flags().Set("all-resources", "true"))
+	}
+	if test.shouldPassUserConfig {
+		require.NoError(t, cmd.Flags().Set("diff-config", path.Join(test.getTestDir(), userConfigFileName)))
+	}
+	resourcesDir := path.Join(test.getTestDir(), resourceDirName)
+	switch mode {
+	case Local:
+		require.NoError(t, cmd.Flags().Set("filename", resourcesDir))
+		require.NoError(t, cmd.Flags().Set("recursive", "true"))
+		break
+	case Live:
+		discoveryResources, resources := getResources(t, resourcesDir)
+		updateTestDiscoveryClient(tf, discoveryResources)
+		setClient(t, resources, tf)
+		break
```

Comment:
Thought `envtest` library is a good one to flag...not sure if it's worth it here but something for internal testing env down the road... https://book.kubebuilder.io/reference/envtest.html